executor/containerdexecutor/executor.go

```diff
@@ -31,6 +32,8 @@ type containerdExecutor struct {
 	networkProviders map[pb.NetMode]network.Provider
 	cgroupParent     string
 	dnsConfig        *oci.DNSConfig
+	running          map[string]chan error
+	mu               *sync.Mutex
```

Comment:
Don't see why this needs to be a pointer

Reply:
Yeah, it was b/c we have `func (w containerdExecutor) Run` instead of taking a pointer type, so that would result in copies of the sync.Mutex.  I can just change the function to accept a `*containerdExecutor` instead.

Reply:
Yes, `*containerdExecutor` is better